docs/providers.md

```diff
@@ -0,0 +1,3 @@
+# Providers
+
+Providers in Goose mean "LLM providers" that Goose can interact with. Providers are defined in the Exchange library for the most part, but you can define your own. 
```

Comment:
Do we want to list the providers here and also mention that you would define your own in a goose-plugins repo?

Reply:
I prefer to decouple users from Exchange as that's an impl detail, so second 👍 on inlining provider support